server/src/main/java/org/elasticsearch/monitor/Probes.java

```
@@ -9,21 +9,29 @@
 
 package org.elasticsearch.monitor;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 import java.lang.management.OperatingSystemMXBean;
 import java.lang.reflect.Method;
 
 public class Probes {
+    private static final Logger logger = LogManager.getLogger(Probes.class);
+
     public static short getLoadAndScaleToPercent(Method method, OperatingSystemMXBean osMxBean) {
+        logger.debug("Starting probe of method {} on osMxBean {}", method, osMxBean);
         if (method != null) {
             try {
                 double load = (double) method.invoke(osMxBean);
                 if (load >= 0) {
                     return (short) (load * 100);
                 }
             } catch (Exception e) {
+                logger.info(() -> "failed to invoke method [" + method + "] on osMxBean [" + osMxBean + "]", e);
                 return -1;
             }
         }
+        logger.debug("Method is null. Returning default value.");
```

I think we should also debug log the exception swallowed in `OsProbe.getMethod`. That will explain _why_ the method is null.

---

And I guess `ProcessProbe.getMethod` and `getUnixMethod` while we're at it (even though `OsProbe` was the one we cared about in this instance). Oh, and a quick grep suggests there are more instances in `SunThreadInfo`, `JvmInfo`, and `HotThreads`. Sorry for the scope creep, but hopefully it's pretty straightforward to just do the same thing in each of those?

---

No you're right :-) Might as well while we are in here!

---

Added additional coverage 👍🏻

---

Thanks. (You've obviously already spotted that `HotThreads` was a false positive in my too-quick quick grep!)